main.c

```diff
@@ -1078,15 +1078,17 @@ int __attribute__((used)) main(void) {
     int exit_code = PYEXEC_FORCED_EXIT;
     bool skip_repl = true;
     bool simulate_reset = true;
+    bool first_run = true;
```

Comment:
`simulate_reset` is now not used at all. I think perhaps an alternative fix is not to add `first_run` back, but to initialize `bool simulate_reset = false` instead of `true`.

Reply:
You are right that doing it with `simulate_reset` is better.  However, just initializing it to `false` is not correct I think, since it would also print `soft reboot` at first power up.

I looked a little deeper into the code and I think I have the right understanding of the behavior now.  The basic problem was that `simulate_reset` was initialized to `true` (correctly), but it never was set to `false` anywhere.  Inside `run_code_py`, it only ever gets set to `true` as well (to make a sleep look like coming out of reset it seems).

I just pushed a new fix that uses `simulate_reset` as requested, but sets it to `false` after the check.  Then it can possibly be set back to `true` inside `run_code_py` when coming out of sleep or stay `false` at every ^D.

Reply:
> However, just initializing it to `false` is not correct I think, since it would also print `soft reboot` at first power up.

Yes, thanks for pointing that out.

The sleep / fake-sleep logic was made a lot simpler, and we missed this case when we redid it.